manifests/mod/ssl.pp

```diff
@@ -174,6 +178,46 @@
     include apache::mod::socache_shmcb
   }
 
+  file { $apache::params::puppet_ssl_dir:
```

Comment:
Why refer to `apache::params` instead of `apache`?

Reply:
Because it is defined only there? It is NOT a parameter in init.pp.
Maybe i misunderstand something about variable scoping and you can explain?

Reply:
Technically `apache` inherits from `apache::params` so you can use `$apache::puppet_ssl_dir`. Then in the future you can easily make it a parameter.

I've seen that this confuses people and isn't really obvious to people.

Reply:
I am still not convinced. I've seen a lot of other places, where apache::params::* variables are used.
It still can be changed, when it shall be come a parameter of the init class.
It would rather confuse me the other way around if i would use $apache::puppet_ssl_dir and don't have a parameter for that. I don't see the need for a parameter.

Reply:
I don't have an opinion on which direction to go here, but do understand that many parts of this module are older than what anyone would call "best practice" today.

Reply:
@DavidS Well, i think we then need a maintainer of this module to decide.

Reply:
When in doubt, I go with what's there in the codebase and the opinion of them who does the work, unless there is a critical overriding factor.

Reply:
>  It would rather confuse me the other way around if i would use $apache::puppet_ssl_dir and don't have a parameter for that. I don't see the need for a parameter.

That's what I've also seen from others. They assume it doesn't work because there's no parameter and forget the inheritance. Implementation wise it doesn't make a difference but code clarity would be a good reason to use `::params::`. I just wanted to highlight it is an option.

Reply:
Thanks for highlighting. I think i leave it as it is, unless the maintainers of the module want a change.